app/models/concerns/foreman/event_subscribers/observable.rb

```diff
@@ -0,0 +1,36 @@
+module Foreman
+  module EventSubscribers
+    module Observable
+      extend ActiveSupport::Concern
+
+      included do
+        class_attribute :event_subscription_hooks
+        self.event_subscription_hooks ||= []
+      end
+
+      module ClassMethods
+        def notify_event_observers(on:, with:, payload: nil, &blk)
+          event_name = "#{event_subscription_namespace}_#{with}"
+          self.event_subscription_hooks |= [event_name]
+          after_commit on: on do
+            actual_payload = blk&.call(self) || payload || { id: id }
+            do_notify_event_observers(with, payload: actual_payload)
+          end
+        end
+
+        def event_subscription_namespace
+          name.underscore
```

Comment:
Should this be `table_name` rather than `name`? Need to test.

Reply:
I'm wondering if it should be prefixed with `model_` or `model.`